browser/js/user/user.factory.js

```diff
@@ -26,5 +24,10 @@ app.factory('UserFactory', function($state, $http){
         })
     }
 
+    UserFactory.autocomplete = function(str){
+        $http.get('/api/user/?username='+str)
```

Comment:
Two things. 1) es6 template strings are simply delightful. Here you could do `` `/api/user/?username=${str}` `` instead of `'/api/user/?username='+str`. 2) `$http.get` has a way of converting an object to url query params, like so: `$http.get('/api/user/', {params: {username: str}})`. See [here](https://docs.angularjs.org/api/ng/service/$http#usage) for more.